jetbrains-core/tst/software/aws/toolkits/jetbrains/services/dynamo/CustomDynamoPaginatorsTest.kt

```diff
@@ -0,0 +1,60 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.dynamo
+
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.Test
+import org.mockito.kotlin.any
+import org.mockito.kotlin.argumentCaptor
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient
+import software.amazon.awssdk.services.dynamodb.model.ExecuteStatementRequest
+import software.amazon.awssdk.services.dynamodb.model.ExecuteStatementResponse
+import software.aws.toolkits.core.utils.delegateMock
+import software.aws.toolkits.jetbrains.services.dynamo.DynamoUtils.executeStatementPaginator
+
+class CustomDynamoPaginatorsTest {
+    @Test
+    fun `execute statement paginator works`() {
+        val statement = "statement"
+
+        val mockClient = delegateMock<DynamoDbClient> {
+            on { executeStatement(any<ExecuteStatementRequest>()) }
+                .thenReturn(ExecuteStatementResponse.builder().nextToken("token").build())
+                .thenReturn(ExecuteStatementResponse.builder().nextToken("token2").build())
+                .thenReturn(ExecuteStatementResponse.builder().build())
+        }
+
+        assertThat(mockClient.executeStatementPaginator(ExecuteStatementRequest.builder().statement(statement).build()).toList()).hasSize(3)
+
+        argumentCaptor<ExecuteStatementRequest> {
+            verify(mockClient, times(3)).executeStatement(capture())
+
+            assertThat(firstValue.nextToken()).isNull()
+            assertThat(firstValue.statement()).isEqualTo(statement)
+
+            assertThat(secondValue.nextToken()).isEqualTo("token")
+            assertThat(secondValue.statement()).isEqualTo(statement)
+
+            assertThat(thirdValue.nextToken()).isEqualTo("token2")
+            assertThat(thirdValue.statement()).isEqualTo(statement)
+        }
+    }
+
+    @Test
+    fun `execute statement paginator bubbles errors`() {
```

Comment:
nit: could do a test where there is 1 good return and one error (but in this case doesn't test a different codepath)

Reply:
So 3 pages total?

Initial -> p2 (good) -> p3 (bad)?